blockchain/blockchain.go

```diff
@@ -394,18 +425,20 @@ func storeStateUpdate(txn db.Transaction, blockNumber uint64, update *core.State
 	return nil
 }
 
-func getStateUpdateByNumber(txn db.Transaction, blockNumber uint64) (update *core.StateUpdate, err error) {
+func getStateUpdateByNumber(txn db.Transaction, blockNumber uint64) (*core.StateUpdate, error) {
 	numBytes := make([]byte, lenOfByteSlice)
 	binary.BigEndian.PutUint64(numBytes, blockNumber)
 
+	update := new(core.StateUpdate)
 	return update, txn.Get(db.StateUpdatesByBlockNumber.Key(numBytes), func(val []byte) error {
```

Comment:
Let's check error explicitly and `return nil, err`; so that we dont return a non-nil `update` when `err != nil`